### PR TITLE
test: increase coverage for shap/explainers/other/_random.py from 29% to 100%

### PR DESCRIPTION
**Pull request by avi-exe16**
## Overview

Closes #3690

This PR increases test coverage for `shap/explainers/other/_random.py` from 29% to 100% by adding a dedicated test file `tests/explainers/other/test_random.py`.

Tests added cover:
- Basic explainer initialization
- Output shape validation
- Non-constant random attributions
- Small value range verification
- Custom call arguments
- Numpy array clustering
- Callable clustering
- Invalid clustering raises NotImplementedError

## Checklist

- [x] All pre-commit checks pass.
- [x] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

File: tests/explainers/other/test_random.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import pytest
+from sklearn.linear_model import LinearRegression
+
+from shap.explainers.other._random import Random
+from shap.maskers import Independent
+
+
+@pytest.fixture
+def model_and_data():
+    X = np.random.randn(50, 4)
```
Comment (naxatra2) [inline tests/explainers/other/test_random.py]: I was working on my own PR in #4384. where the maintainer suggested some changes in this line maybe you should also look into it. 

Adding seed fixes is a good idea.

Comment (avi-exe16) [reply tests/explainers/other/test_random.py]: Thanks for the feedback ,  I have fixed the seeds using np.random.default_rng(42) for reproducibility.
